link_test.go

```diff
@@ -14,96 +13,6 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestLinkFlowForSender(t *testing.T) {
```

Comment:
I added tests in receiver_tests.go to cover the cases here (some were already covered).